crates/db_views/site/src/impls.rs

```diff
@@ -93,3 +108,93 @@ pub async fn user_backup_list_to_user_settings_backup(
     discussion_languages,
   })
 }
+
+impl FederatedInstancesView {
+  pub async fn list(pool: &mut DbPool<'_>) -> LemmyResult<Vec<Self>> {
+    let conn = &mut get_conn(pool).await?;
+    instance::table
+      // omit instance representing the local site
+      .left_join(site::table.left_join(local_site::table))
+      .filter(local_site::id.is_null())
+      .left_join(federation_blocklist::table)
+      .left_join(federation_allowlist::table)
+      .left_join(federation_queue_state::table)
+      // Show recently updated instances and those with valid metadata first
+      .order((
+        instance::updated_at.desc(),
+        instance::software.asc().nulls_last(),
+      ))
+      .select(Self::as_select())
+      .get_results(conn)
+      .await
+      .with_lemmy_type(LemmyErrorType::NotFound)
+  }
+}
```

Comment:
Here is the main change. Probably needs pagination, eg lemmy.ml has 10k linked instances. It would also make sense to add a parameter `type: "Linked" | "Allowed" | "Blocked"`.